Project group name is wrong

### Expected Behavior

A working starter app using the feature `json-schema` when running
`mn create-app --build=gradle_kotlin --jdk=21 --lang=java --test=junit --features=json-schema com.example.json-schema`

### Actual Behaviour

Micronaut Launch/create-app does not work because the name of the project group is wrong.

Micronaut Launch uses `.json-schema`, while the actual name of the dependency is `.jsonschema`.
`implementation("io.micronaut.jsonschema:micronaut-json-schema-annotations")`
vs
`implementation("io.micronaut.json-schema:micronaut-json-schema-annotations")`

Based on what the naming convention seems to be for Micronaut modules, this group should be called `json-schema`?

Either way, the starter is not working.

### Steps To Reproduce

1. run `mn create-app --build=gradle_kotlin --jdk=21 --lang=java --test=junit --features=json-schema com.example.json-schema`
2. Build and observe the error
```
FAILURE: Build failed with an exception.

* What went wrong:
Could not determine the dependencies of task ':optimizedJitJarAll'.
> Could not resolve all dependencies for configuration ':runtimeClasspath'.
   > Could not find io.micronaut.json-schema:micronaut-json-schema-annotations:.
     Required by:
         project :
```

### Environment Information

- macOS Sonoma 14.5
- Eclipse temurin-21

### Example Application

_No response_

### Version

4.5.0

--- a/starter-core/src/main/java/io/micronaut/starter/build/dependencies/MicronautDependencyUtils.java
+++ b/starter-core/src/main/java/io/micronaut/starter/build/dependencies/MicronautDependencyUtils.java
@@ -42,7 +42,7 @@ public final class MicronautDependencyUtils {
     public static final String GROUP_ID_MICRONAUT_CRAC = "io.micronaut.crac";
     public static final String GROUP_ID_MICRONAUT_ECLIPSESTORE = "io.micronaut.eclipsestore";
     public static final String GROUP_ID_MICRONAUT_GCP = "io.micronaut.gcp";
-    public static final String GROUP_ID_MICRONAUT_JSON_SCHEMA = "io.micronaut.json-schema";
+    public static final String GROUP_ID_MICRONAUT_JSON_SCHEMA = "io.micronaut.jsonschema";
     public static final String GROUP_ID_MICRONAUT_KAFKA = "io.micronaut.kafka";
     public static final String GROUP_ID_MICRONAUT_OCI = "io.micronaut.oraclecloud";
     public static final String GROUP_ID_MICRONAUT_OPENSEARCH = "io.micronaut.opensearch";

--- a/starter-core/src/test/groovy/io/micronaut/starter/feature/build/MicronautAotSpec.groovy
+++ b/starter-core/src/test/groovy/io/micronaut/starter/feature/build/MicronautAotSpec.groovy
@@ -167,7 +167,9 @@ class MicronautAotSpec extends ApplicationContextSpec implements CommandOutputFi
 
         expect:
         output.containsKey('aot-jar.properties')
-        output["aot-jar.properties"].trim() == expected.trim()
+
+        def aotconfig = output["aot-jar.properties"]
+        aotconfig.trim() == expected.trim()
         !output.containsKey('aot-native-image.properties')
     }
 

--- a/starter-core/src/test/groovy/io/micronaut/starter/feature/json/JsonSchemaFeatureSpec.groovy
+++ b/starter-core/src/test/groovy/io/micronaut/starter/feature/json/JsonSchemaFeatureSpec.groovy
@@ -63,8 +63,8 @@ class JsonSchemaFeatureSpec extends ApplicationContextSpec implements CommandOut
 
         then:
         template
-        verifier.hasAnnotationProcessor("io.micronaut.json-schema", "micronaut-json-schema-processor")
-        verifier.hasDependency("io.micronaut.json-schema", "micronaut-json-schema-annotations")
+        verifier.hasAnnotationProcessor("io.micronaut.jsonschema", "micronaut-json-schema-processor")
+        verifier.hasDependency("io.micronaut.jsonschema", "micronaut-json-schema-annotations")
 
         where:
         [buildTool, language] << [BuildTool.values(), Language.values()].combinations()

--- a/starter-core/src/test/groovy/io/micronaut/starter/feature/json/JsonSchemaValidationFeatureSpec.groovy
+++ b/starter-core/src/test/groovy/io/micronaut/starter/feature/json/JsonSchemaValidationFeatureSpec.groovy
@@ -53,7 +53,7 @@ class JsonSchemaValidationFeatureSpec extends ApplicationContextSpec implements 
 
         then:
         template
-        verifier.hasDependency("io.micronaut.json-schema", "micronaut-json-schema-validation", Scope.TEST)
+        verifier.hasDependency("io.micronaut.jsonschema", "micronaut-json-schema-validation", Scope.TEST)
 
         where:
         [buildTool, language] << [BuildTool.values(), Language.values()].combinations()

--- a/starter-core/src/test/resources/expected-aot-jar.properties
+++ b/starter-core/src/test/resources/expected-aot-jar.properties
@@ -23,7 +23,7 @@ scan.reactive.types.enabled=true
 # Deduces the environment at build time instead of runtime
 deduce.environment.enabled=true
 
-# Checks of existence of some types at build time instead of runtime
+# Checks for the existence of some types at build time instead of runtime
 known.missing.types.enabled=true
 
 # Precomputes property sources at build time

--- a/starter-core/src/test/resources/expected-aot-native-image.properties
+++ b/starter-core/src/test/resources/expected-aot-native-image.properties
@@ -26,7 +26,7 @@ scan.reactive.types.enabled=true
 # Deduces the environment at build time instead of runtime
 deduce.environment.enabled=true
 
-# Checks of existence of some types at build time instead of runtime
+# Checks for the existence of some types at build time instead of runtime
 known.missing.types.enabled=true
 
 # Precomputes property sources at build time

